source/fundamentals/aggregation/aggregation-expression-operations.txt

```diff
@@ -282,6 +282,107 @@ the Query API:
      }
    ]
 
+Function Operations
+~~~~~~~~~~~~~~~~~~~
+
+You can apply custom functions to values of type
+``MqlValue`` using the methods described in this section.
+
+To improve readability and brevity in your code, you can move redundant code into
+static methods. However, it is often not possible to chain these methods in
+Java. The ``passTo()`` method lets you chain custom methods in your code.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 40 60
+
+   * - Java Method
+     - Aggregation Pipeline Operator
+
+   * - `passTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#passTo(java.util.function.Function)>`__
+       
+       | **Typed Variants**:
+       | `passArrayTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#passArrayTo(java.util.function.Function)>`__
+       | `passBooleanTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlBoolean.html#passBooleanTo(java.util.function.Function)>`__
+       | `passDateTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDate.html#passDateTo(java.util.function.Function)>`__
+       | `passDocumentTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#passDocumentTo(java.util.function.Function)>`__
+       | `passIntegerTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlInteger.html#passIntegerTo(java.util.function.Function)>`__
+       | `passMapTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlMap.html#passMapTo(java.util.function.Function)>`__
+       | `passNumberTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlNumber.html#passNumberTo(java.util.function.Function)>`__
+       | `passStringTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#passStringTo(java.util.function.Function)>`__
+
+     - :manual:`$function </reference/operator/aggregation/function/>`
+
+.. include:: /includes/fundamentals/agg-mql-diff.rst
+
+Suppose you need to determine how a class is performing against some
+benchmarks. You want to find the average final grade for each class and
+compare it against the benchmark values.
+
+To calculate each class average, write an averaging method called
+``classAvg()`` and pass the ``students`` array to this method by using
+the ``passArrayTo()`` method. To compare each class average against your
+defined benchmarks, write a method called ``evalAvg()``. Apply this
+method to the class averages by chaining the ``passNumberTo()`` method
+to the previous method. To output the result as the ``evaluation``
+field, use the ``project()`` method.
```

Comment:
It seems worth saying that the two custom methods can be reused in other queries (for example, you might write a query that filters certain students not by class, but perhaps by entry year, or teacher; and you might use the evaluate method to evaluate a single student).

Reply:
Looks good